src/modules/sched-simple/rlist.c

```diff
@@ -85,9 +85,24 @@ static struct rnode *rlist_find_rank (struct rlist *rl, uint32_t rank)
 
 static int idset_cmp (struct idset *set1, struct idset *set2)
 {
-    if (idset_equal (set1, set2))
+    int rv = 0;
+    if (idset_equal (set1, set2)) {
         return 0;
-    return idset_count (set1) - idset_count (set2);
+    }
+    /*  try sorting first on idset count:
+     */
+    rv = idset_count (set1) - idset_count (set2);
+    if (rv == 0) {
+        /*  idset size is the same, sort by first differing integer in set:
+         */
+        unsigned int a = idset_first (set1);
+        unsigned int b = idset_first (set2);
+        while ((rv = (a - b)) == 0) {
+            a = idset_next (set1, a);
+            b = idset_next (set1, b);
```

Comment:
should be `idset_next (set2)`?

Reply:
yep, thanks!

Reply:
might have to quickly figure out how to ensure that code gets exercised, but I may not get to it...

Reply:
@chu11, quickly fixed this issue with a new commit. Once you're done reviewing, I'll squash.